nexus/mgs-updates/src/driver_update.rs

```diff
@@ -599,6 +630,36 @@ enum UpdateWaitError {
     Indeterminate(#[source] PrecheckError),
 }
 
+// Timeouts, timeouts: always wrong!
+//
+// We have to pick some maximum time we're willing to wait for the `post_update`
+// hook to complete. In general this hook is responsible for resetting the
+// updated target to cause it to boot into its new version, but the details vary
+// wildly by device type (e.g., resetting the RoT requires multiple resets) and
+// the expected amount of time also varies wildly (e.g., resetting a gimlet SP
+// takes a few seconds, resetting a sidecar SP can take 10s of seconds,
+// resetting a sled after a host OS update takes minutes).
+fn post_update_timeout(update: &PendingMgsUpdate) -> Duration {
+    match &update.details {
+        PendingMgsUpdateDetails::Sp { .. } => {
+            // We're resetting an SP; use a generous timeout for sleds and power
+            // shelf controllers (which should take a few seconds) and an even
+            // more generaous timeout for switches (which we've seen take 10-20
+            // seconds in practice).
+            match update.sp_type {
+                SpType::Sled | SpType::Power => Duration::from_secs(60),
+                SpType::Switch => Duration::from_secs(120),
+            }
+        }
+        PendingMgsUpdateDetails::Rot { .. }
+        | PendingMgsUpdateDetails::RotBootloader { .. } => {
+            // Resetting the RoT and the bootloader should be quick (a few
+            // seconds each).
+            Duration::from_secs(60)
```

Comment:
I think for the bootloader we'll want more than 60 seconds. We're giving ourselves 120 seconds for boot info to show up after the first signature check reset, and then we set the active slot and another reset after that

https://github.com/oxidecomputer/omicron/blob/3b3931b35303735820cae673782f3a9d7fc27a80/nexus/mgs-updates/src/rot_bootloader_updater.rs#L31

https://github.com/oxidecomputer/omicron/blob/3b3931b35303735820cae673782f3a9d7fc27a80/nexus/mgs-updates/src/rot_bootloader_updater.rs#L191-L235

Reply:
Changed back to 120 in [f37d46a](https://github.com/oxidecomputer/omicron/pull/8736/commits/f37d46af5aaf4bf69a04106fd5d0af1733375d37)